app/models/concerns/profile_tailoring.rb

```diff
@@ -33,4 +33,20 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def tailoring_valid?
+    rule_groups = RuleGroup.where(id: rules.includes(:rule_group)
+                           .references(:rule_group).flat_map { |r| r.rule_group&.path_ids })
+
+    conflicts, requires = %w[conflicts requires].map do |relationship|
+      with_relationships(relationship, rules, rule_groups).map(&:right)
+    end
+
+    valid_relationships?(rules, rule_groups, requires, conflicts)
+  end
+
+  def valid_relationships?(rules, rule_groups, requires, conflicts)
+    rules_and_groups = rules | rule_groups
+    (requires - rules_and_groups).empty? && (conflicts - rules_and_groups) == conflicts
```

Comment:
Not tested, but should be simpler:
```suggestion
    (requires - rules_and_groups).empty? && (conflicts & rules_and_groups).empty?
```